skfp/descriptors/topological.py

```diff
@@ -218,9 +404,46 @@ def zagreb_index(mol: Mol) -> int:
     Examples
     --------
     >>> from rdkit.Chem import MolFromSmiles
-    >>> from skfp.descriptors.topological import zagreb_index
+    >>> from skfp.descriptors.topological import zagreb_index_m1
     >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
-    >>> zagreb_index(mol)
+    >>> zagreb_index_m1(mol)
     24
     """
     return int(sum(atom.GetDegree() ** 2 for atom in mol.GetAtoms()))
+
+
+def zagreb_index_m2(mol: Mol) -> int:
+    """
+    Second Zagreb Index.
+
+    Calculates the Second Zagreb Index [1]_, defined as the sum of the product of
+    degrees of all pairs of adjacent atoms in the molecule. It provides a measure
+    related to molecular branching and the connectivity of bonds.
```

Comment:
Add a formula, it's simple if you write it in terms of bonds, i.e. use `e=(u,v)` in the product subscript